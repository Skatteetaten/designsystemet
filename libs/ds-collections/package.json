{
  "name": "@skatteetaten/ds-collections",
  "groupId": "no.skatteetaten.aurora",
  "version": "1.4.0",
  "license": "Apache-2.0",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^23",
    "react": "^18",
    "react-i18next": "^14",
    "@skatteetaten/ds-core-utils": "1.3.0",
    "@skatteetaten/ds-icons": "1.3.0",
    "@skatteetaten/ds-buttons": "1.3.0",
    "@skatteetaten/ds-content": "1.3.0",
    "@skatteetaten/ds-typography": "1.3.0",
    "@skatteetaten/ds-progress": "1.3.0"
  },
  "scripts": {}
}
