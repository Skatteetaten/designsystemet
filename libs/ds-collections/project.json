{
  "name": "ds-collections",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-collections/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"]
    },
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-collections",
        "tsConfig": "libs/ds-collections/tsconfig.lib.json",
        "project": "libs/ds-collections/package.json",
        "entryFile": "libs/ds-collections/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-collections/rollup.config.js"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-collections/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/ds-collections/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-collections/**/*.css",
          "libs/ds-collections/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      },
      "dependsOn": [
        {
          "projects": "ds-core-designtokens",
          "target": "build"
        }
      ]
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-collections"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-collections {args.switches}",
        "cwd": ""
      }
    }
  }
}
