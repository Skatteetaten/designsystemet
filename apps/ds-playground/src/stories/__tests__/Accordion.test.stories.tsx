import { Meta, StoryFn, StoryObj } from '@storybook/react';
import { expect, within } from '@storybook/test';

import { Accordion, AccordionProps } from '@skatteetaten/ds-collections';

const elementId = 'htmlId';
const defaultTitle = 'Meg selv';
const defaultSubtitle = 'Beskrivelse';
const defaultContent =
  'Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.';

const meta = {
  component: Accordion,
  title: 'Tester/Accordion/Accordion',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: { table: { disable: true } },
    color: { table: { disable: true } },
    size: { table: { disable: true } },
    iconPosition: { table: { disable: true } },
  },
  args: {
    children: (
      <Accordion.Item title={defaultTitle}>{defaultContent}</Accordion.Item>
    ),
  },
} satisfies Meta<typeof Accordion>;
export default meta;
type Story = StoryObj<typeof meta>;

const TemplateWithAllColors: StoryFn<typeof Accordion> = () => {
  const items = [
    <Accordion.Item key={'i1'} title={defaultTitle} subtitle={defaultSubtitle}>
      {defaultContent}
    </Accordion.Item>,
    <Accordion.Item key={'i2'} title={defaultTitle} subtitle={defaultSubtitle}>
      {defaultContent}
    </Accordion.Item>,
  ];

  return (
    <>
      <Accordion className={'bottomSpacingXL'} color={'none'}>
        {items}
      </Accordion>
      <Accordion className={'bottomSpacingXL'} color={'burgundy'}>
        {items}
      </Accordion>
      <Accordion className={'bottomSpacingXL'} color={'denim'}>
        {items}
      </Accordion>

      <Accordion className={'bottomSpacingXL'} color={'forest'}>
        {items}
      </Accordion>
      <Accordion className={'bottomSpacingXL'} color={'graphite'}>
        {items}
      </Accordion>
      <Accordion className={'bottomSpacingXL'} color={'ochre'}>
        {items}
      </Accordion>
    </>
  );
};

const defaultArgs: AccordionProps = {
  children: (
    <Accordion.Item title={defaultTitle}>{defaultContent}</Accordion.Item>
  ),
};

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLDivElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getAllByRole('generic')[1];
    await expect(container).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes(FA2-5)',
  args: {
    ...defaultArgs,
    id: elementId,
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getAllByRole('generic')[1];
    await expect(container).toHaveClass('dummyClassname');
    await expect(container).toHaveAttribute('id', elementId);
    await expect(container).toHaveAttribute('lang', 'nb');
    await expect(container).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

export const Defaults = {
  name: 'Defaults (A2, A4, A7 delvis)',
  args: defaultArgs,
} satisfies Story;

export const WithIconLeft = {
  name: 'With Icon Left (A2)',
  args: {
    iconPosition: 'left',
    ...defaultArgs,
  },
} satisfies Story;

export const WithAllColorVariants = {
  render: TemplateWithAllColors,
  name: 'With All Color Variants (A1)',
  args: defaultArgs,
} satisfies Story;

export const WithSmall = {
  name: 'Variant Small (A7)',
  args: { size: 'small', ...defaultArgs },
} satisfies Story;

export const WithMedium = {
  name: 'Variant Medium (A7)',
  args: { size: 'medium', ...defaultArgs },
} satisfies Story;

export const WithLarge = {
  name: 'Variant Large (A7)',
  args: { size: 'large', ...defaultArgs },
} satisfies Story;
