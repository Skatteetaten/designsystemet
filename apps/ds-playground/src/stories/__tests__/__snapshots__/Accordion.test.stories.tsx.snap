// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Accordion/Accordion Defaults smoke-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="_accordionItem_ihaf2_1">
      <div class="_tag_ihaf2_11">
        <button class="_header_ihaf2_14  _header_medium_ihaf2_71"
                aria-expanded="false"
                type="button"
        >
          <div class="_titleWrapper_ihaf2_76">
            <div class="_title_ihaf2_76 _title_medium_ihaf2_100">
              Meg selv
            </div>
          </div>
          <div class="_iconWrapper_ihaf2_80 _iconWrapper_medium_ihaf2_86">
            <svg class="_systemIcon_dry7n_2 _systemIcon_large_dry7n_15 _icon_ihaf2_18 _chevron_medium_ihaf2_29 _chevron_closed_ihaf2_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M18.1 7.98001L12 14.08L5.9 7.98001L4 9.88L12 17.88L20 9.88L18.1 7.98001Z">
              </path>
            </svg>
          </div>
        </button>
      </div>
      <div class="_content_collapsed_ihaf2_120">
        Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/Accordion/Accordion WithAllColorVariants smoke-test 1`] = `
<div data-test-block="true">
  <div class="bottomSpacingXL">
    <div class="_accordionItem_ihaf2_1">
      <div class="_tag_ihaf2_11">
        <button class="_header_ihaf2_14  _header_medium_ihaf2_71"
                aria-expanded="false"
                type="button"
        >
          <div class="_titleWrapper_ihaf2_76">
            <div class="_title_ihaf2_76 _title_medium_ihaf2_100">
              Meg selv
            </div>
            <p class="_subtitle_ihaf2_107">
              Beskrivelse
            </p>
          </div>
          <div class="_iconWrapper_ihaf2_80 _iconWrapper_medium_ihaf2_86">
            <svg class="_systemIcon_dry7n_2 _systemIcon_large_dry7n_15 _icon_ihaf2_18 _chevron_medium_ihaf2_29 _chevron_closed_ihaf2_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M18.1 7.98001L12 14.08L5.9 7.98001L4 9.88L12 17.88L20 9.88L18.1 7.98001Z">
              </path>
            </svg>
          </div>
        </button>
      </div>
      <div class="_content_collapsed_ihaf2_120">
        Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.
      </div>
    </div>
    <div class="_accordionItem_ihaf2_1">
      <div class="_tag_ihaf2_11">
        <button class="_header_ihaf2_14  _header_medium_ihaf2_71"
                aria-expanded="false"
                type="button"
        >
          <div class="_titleWrapper_ihaf2_76">
            <div class="_title_ihaf2_76 _title_medium_ihaf2_100">
              Meg selv
            </div>
            <p class="_subtitle_ihaf2_107">
              Beskrivelse
            </p>
          </div>
          <div class="_iconWrapper_ihaf2_80 _iconWrapper_medium_ihaf2_86">
            <svg class="_systemIcon_dry7n_2 _systemIcon_large_dry7n_15 _icon_ihaf2_18 _chevron_medium_ihaf2_29 _chevron_closed_ihaf2_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M18.1 7.98001L12 14.08L5.9 7.98001L4 9.88L12 17.88L20 9.88L18.1 7.98001Z">
              </path>
            </svg>
          </div>
        </button>
      </div>
      <div class="_content_collapsed_ihaf2_120">
        Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.
      </div>
    </div>
  </div>
  <div class="_accordion_burgundy_1hopi_14 bottomSpacingXL">
    <div class="_accordionItem_ihaf2_1">
      <div class="_tag_ihaf2_11">
        <button class="_header_ihaf2_14  _header_medium_ihaf2_71"
                aria-expanded="false"
                type="button"
        >
          <div class="_titleWrapper_ihaf2_76">
            <div class="_title_ihaf2_76 _title_medium_ihaf2_100">
              Meg selv
            </div>
            <p class="_subtitle_ihaf2_107">
              Beskrivelse
            </p>
          </div>
          <div class="_iconWrapper_ihaf2_80 _iconWrapper_medium_ihaf2_86">
            <svg class="_systemIcon_dry7n_2 _systemIcon_large_dry7n_15 _icon_ihaf2_18 _chevron_medium_ihaf2_29 _chevron_closed_ihaf2_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M18.1 7.98001L12 14.08L5.9 7.98001L4 9.88L12 17.88L20 9.88L18.1 7.98001Z">
              </path>
            </svg>
          </div>
        </button>
      </div>
      <div class="_content_collapsed_ihaf2_120">
        Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.
      </div>
    </div>
    <div class="_accordionItem_ihaf2_1">
      <div class="_tag_ihaf2_11">
        <button class="_header_ihaf2_14  _header_medium_ihaf2_71"
                aria-expanded="false"
                type="button"
        >
          <div class="_titleWrapper_ihaf2_76">
            <div class="_title_ihaf2_76 _title_medium_ihaf2_100">
              Meg selv
            </div>
            <p class="_subtitle_ihaf2_107">
              Beskrivelse
            </p>
          </div>
          <div class="_iconWrapper_ihaf2_80 _iconWrapper_medium_ihaf2_86">
            <svg class="_systemIcon_dry7n_2 _systemIcon_large_dry7n_15 _icon_ihaf2_18 _chevron_medium_ihaf2_29 _chevron_closed_ihaf2_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M18.1 7.98001L12 14.08L5.9 7.98001L4 9.88L12 17.88L20 9.88L18.1 7.98001Z">
              </path>
            </svg>
          </div>
        </button>
      </div>
      <div class="_content_collapsed_ihaf2_120">
        Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.
      </div>
    </div>
  </div>
  <div class="_accordion_denim_1hopi_8 bottomSpacingXL">
    <div class="_accordionItem_ihaf2_1">
      <div class="_tag_ihaf2_11">
        <button class="_header_ihaf2_14  _header_medium_ihaf2_71"
                aria-expanded="false"
                type="button"
        >
          <div class="_titleWrapper_ihaf2_76">
            <div class="_title_ihaf2_76 _title_medium_ihaf2_100">
              Meg selv
            </div>
            <p class="_subtitle_ihaf2_107">
              Beskrivelse
            </p>
          </div>
          <div class="_iconWrapper_ihaf2_80 _iconWrapper_medium_ihaf2_86">
            <svg class="_systemIcon_dry7n_2 _systemIcon_large_dry7n_15 _icon_ihaf2_18 _chevron_medium_ihaf2_29 _chevron_closed_ihaf2_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M18.1 7.98001L12 14.08L5.9 7.98001L4 9.88L12 17.88L20 9.88L18.1 7.98001Z">
              </path>
            </svg>
          </div>
        </button>
      </div>
      <div class="_content_collapsed_ihaf2_120">
        Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.
      </div>
    </div>
    <div class="_accordionItem_ihaf2_1">
      <div class="_tag_ihaf2_11">
        <button class="_header_ihaf2_14  _header_medium_ihaf2_71"
                aria-expanded="false"
                type="button"
        >
          <div class="_titleWrapper_ihaf2_76">
            <div class="_title_ihaf2_76 _title_medium_ihaf2_100">
              Meg selv
            </div>
            <p class="_subtitle_ihaf2_107">
              Beskrivelse
            </p>
          </div>
          <div class="_iconWrapper_ihaf2_80 _iconWrapper_medium_ihaf2_86">
            <svg class="_systemIcon_dry7n_2 _systemIcon_large_dry7n_15 _icon_ihaf2_18 _chevron_medium_ihaf2_29 _chevron_closed_ihaf2_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M18.1 7.98001L12 14.08L5.9 7.98001L4 9.88L12 17.88L20 9.88L18.1 7.98001Z">
              </path>
            </svg>
          </div>
        </button>
      </div>
      <div class="_content_collapsed_ihaf2_120">
        Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.
      </div>
    </div>
  </div>
  <div class="_accordion_forest_1hopi_5 bottomSpacingXL">
    <div class="_accordionItem_ihaf2_1">
      <div class="_tag_ihaf2_11">
        <button class="_header_ihaf2_14  _header_medium_ihaf2_71"
                aria-expanded="false"
                type="button"
        >
          <div class="_titleWrapper_ihaf2_76">
            <div class="_title_ihaf2_76 _title_medium_ihaf2_100">
              Meg selv
            </div>
            <p class="_subtitle_ihaf2_107">
              Beskrivelse
            </p>
          </div>
          <div class="_iconWrapper_ihaf2_80 _iconWrapper_medium_ihaf2_86">
            <svg class="_systemIcon_dry7n_2 _systemIcon_large_dry7n_15 _icon_ihaf2_18 _chevron_medium_ihaf2_29 _chevron_closed_ihaf2_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M18.1 7.98001L12 14.08L5.9 7.98001L4 9.88L12 17.88L20 9.88L18.1 7.98001Z">
              </path>
            </svg>
          </div>
        </button>
      </div>
      <div class="_content_collapsed_ihaf2_120">
        Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.
      </div>
    </div>
    <div class="_accordionItem_ihaf2_1">
      <div class="_tag_ihaf2_11">
        <button class="_header_ihaf2_14  _header_medium_ihaf2_71"
                aria-expanded="false"
                type="button"
        >
          <div class="_titleWrapper_ihaf2_76">
            <div class="_title_ihaf2_76 _title_medium_ihaf2_100">
              Meg selv
            </div>
            <p class="_subtitle_ihaf2_107">
              Beskrivelse
            </p>
          </div>
          <div class="_iconWrapper_ihaf2_80 _iconWrapper_medium_ihaf2_86">
            <svg class="_systemIcon_dry7n_2 _systemIcon_large_dry7n_15 _icon_ihaf2_18 _chevron_medium_ihaf2_29 _chevron_closed_ihaf2_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M18.1 7.98001L12 14.08L5.9 7.98001L4 9.88L12 17.88L20 9.88L18.1 7.98001Z">
              </path>
            </svg>
          </div>
        </button>
      </div>
      <div class="_content_collapsed_ihaf2_120">
        Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.
      </div>
    </div>
  </div>
  <div class="_accordion_graphite_1hopi_17 bottomSpacingXL">
    <div class="_accordionItem_ihaf2_1">
      <div class="_tag_ihaf2_11">
        <button class="_header_ihaf2_14  _header_medium_ihaf2_71"
                aria-expanded="false"
                type="button"
        >
          <div class="_titleWrapper_ihaf2_76">
            <div class="_title_ihaf2_76 _title_medium_ihaf2_100">
              Meg selv
            </div>
            <p class="_subtitle_ihaf2_107">
              Beskrivelse
            </p>
          </div>
          <div class="_iconWrapper_ihaf2_80 _iconWrapper_medium_ihaf2_86">
            <svg class="_systemIcon_dry7n_2 _systemIcon_large_dry7n_15 _icon_ihaf2_18 _chevron_medium_ihaf2_29 _chevron_closed_ihaf2_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M18.1 7.98001L12 14.08L5.9 7.98001L4 9.88L12 17.88L20 9.88L18.1 7.98001Z">
              </path>
            </svg>
          </div>
        </button>
      </div>
      <div class="_content_collapsed_ihaf2_120">
        Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.
      </div>
    </div>
    <div class="_accordionItem_ihaf2_1">
      <div class="_tag_ihaf2_11">
        <button class="_header_ihaf2_14  _header_medium_ihaf2_71"
                aria-expanded="false"
                type="button"
        >
          <div class="_titleWrapper_ihaf2_76">
            <div class="_title_ihaf2_76 _title_medium_ihaf2_100">
              Meg selv
            </div>
            <p class="_subtitle_ihaf2_107">
              Beskrivelse
            </p>
          </div>
          <div class="_iconWrapper_ihaf2_80 _iconWrapper_medium_ihaf2_86">
            <svg class="_systemIcon_dry7n_2 _systemIcon_large_dry7n_15 _icon_ihaf2_18 _chevron_medium_ihaf2_29 _chevron_closed_ihaf2_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M18.1 7.98001L12 14.08L5.9 7.98001L4 9.88L12 17.88L20 9.88L18.1 7.98001Z">
              </path>
            </svg>
          </div>
        </button>
      </div>
      <div class="_content_collapsed_ihaf2_120">
        Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.
      </div>
    </div>
  </div>
  <div class="_accordion_ochre_1hopi_11 bottomSpacingXL">
    <div class="_accordionItem_ihaf2_1">
      <div class="_tag_ihaf2_11">
        <button class="_header_ihaf2_14  _header_medium_ihaf2_71"
                aria-expanded="false"
                type="button"
        >
          <div class="_titleWrapper_ihaf2_76">
            <div class="_title_ihaf2_76 _title_medium_ihaf2_100">
              Meg selv
            </div>
            <p class="_subtitle_ihaf2_107">
              Beskrivelse
            </p>
          </div>
          <div class="_iconWrapper_ihaf2_80 _iconWrapper_medium_ihaf2_86">
            <svg class="_systemIcon_dry7n_2 _systemIcon_large_dry7n_15 _icon_ihaf2_18 _chevron_medium_ihaf2_29 _chevron_closed_ihaf2_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M18.1 7.98001L12 14.08L5.9 7.98001L4 9.88L12 17.88L20 9.88L18.1 7.98001Z">
              </path>
            </svg>
          </div>
        </button>
      </div>
      <div class="_content_collapsed_ihaf2_120">
        Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.
      </div>
    </div>
    <div class="_accordionItem_ihaf2_1">
      <div class="_tag_ihaf2_11">
        <button class="_header_ihaf2_14  _header_medium_ihaf2_71"
                aria-expanded="false"
                type="button"
        >
          <div class="_titleWrapper_ihaf2_76">
            <div class="_title_ihaf2_76 _title_medium_ihaf2_100">
              Meg selv
            </div>
            <p class="_subtitle_ihaf2_107">
              Beskrivelse
            </p>
          </div>
          <div class="_iconWrapper_ihaf2_80 _iconWrapper_medium_ihaf2_86">
            <svg class="_systemIcon_dry7n_2 _systemIcon_large_dry7n_15 _icon_ihaf2_18 _chevron_medium_ihaf2_29 _chevron_closed_ihaf2_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M18.1 7.98001L12 14.08L5.9 7.98001L4 9.88L12 17.88L20 9.88L18.1 7.98001Z">
              </path>
            </svg>
          </div>
        </button>
      </div>
      <div class="_content_collapsed_ihaf2_120">
        Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/Accordion/Accordion WithAttributes play-test 1`] = `
<div data-test-block="true">
  <div class="dummyClassname"
       lang="nb"
       id="htmlId"
       data-testid="123ID"
  >
    <div class="_accordionItem_ihaf2_1">
      <div class="_tag_ihaf2_11">
        <button class="_header_ihaf2_14  _header_medium_ihaf2_71"
                aria-expanded="false"
                type="button"
        >
          <div class="_titleWrapper_ihaf2_76">
            <div class="_title_ihaf2_76 _title_medium_ihaf2_100">
              Meg selv
            </div>
          </div>
          <div class="_iconWrapper_ihaf2_80 _iconWrapper_medium_ihaf2_86">
            <svg class="_systemIcon_dry7n_2 _systemIcon_large_dry7n_15 _icon_ihaf2_18 _chevron_medium_ihaf2_29 _chevron_closed_ihaf2_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M18.1 7.98001L12 14.08L5.9 7.98001L4 9.88L12 17.88L20 9.88L18.1 7.98001Z">
              </path>
            </svg>
          </div>
        </button>
      </div>
      <div class="_content_collapsed_ihaf2_120">
        Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/Accordion/Accordion WithIconLeft smoke-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="_accordionItem_ihaf2_1">
      <div class="_tag_ihaf2_11">
        <button class="_header_ihaf2_14 _header_iconLeft_ihaf2_67 _header_medium_ihaf2_71"
                aria-expanded="false"
                type="button"
        >
          <div class="_titleWrapper_ihaf2_76">
            <div class="_title_ihaf2_76 _title_medium_ihaf2_100">
              Meg selv
            </div>
          </div>
          <div class="_iconWrapper_ihaf2_80 _iconWrapper_medium_ihaf2_86">
            <svg class="_systemIcon_dry7n_2 _systemIcon_large_dry7n_15 _icon_ihaf2_18 _chevron_medium_ihaf2_29 _chevron_closed_ihaf2_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M18.1 7.98001L12 14.08L5.9 7.98001L4 9.88L12 17.88L20 9.88L18.1 7.98001Z">
              </path>
            </svg>
          </div>
        </button>
      </div>
      <div class="_content_collapsed_ihaf2_120">
        Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/Accordion/Accordion WithLarge smoke-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="_accordionItem_ihaf2_1">
      <div class="_tag_ihaf2_11">
        <button class="_header_ihaf2_14  _header_large_ihaf2_71"
                aria-expanded="false"
                type="button"
        >
          <div class="_titleWrapper_ihaf2_76">
            <div class="_title_ihaf2_76 _title_large_ihaf2_103">
              Meg selv
            </div>
          </div>
          <div class="_iconWrapper_ihaf2_80 _iconWrapper_large_ihaf2_89">
            <svg class="_systemIcon_dry7n_2 _systemIcon_large_dry7n_15 _icon_ihaf2_18 _chevron_large_ihaf2_32 _chevron_closed_ihaf2_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M18.1 7.98001L12 14.08L5.9 7.98001L4 9.88L12 17.88L20 9.88L18.1 7.98001Z">
              </path>
            </svg>
          </div>
        </button>
      </div>
      <div class="_content_collapsed_ihaf2_120">
        Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/Accordion/Accordion WithMedium smoke-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="_accordionItem_ihaf2_1">
      <div class="_tag_ihaf2_11">
        <button class="_header_ihaf2_14  _header_medium_ihaf2_71"
                aria-expanded="false"
                type="button"
        >
          <div class="_titleWrapper_ihaf2_76">
            <div class="_title_ihaf2_76 _title_medium_ihaf2_100">
              Meg selv
            </div>
          </div>
          <div class="_iconWrapper_ihaf2_80 _iconWrapper_medium_ihaf2_86">
            <svg class="_systemIcon_dry7n_2 _systemIcon_large_dry7n_15 _icon_ihaf2_18 _chevron_medium_ihaf2_29 _chevron_closed_ihaf2_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M18.1 7.98001L12 14.08L5.9 7.98001L4 9.88L12 17.88L20 9.88L18.1 7.98001Z">
              </path>
            </svg>
          </div>
        </button>
      </div>
      <div class="_content_collapsed_ihaf2_120">
        Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/Accordion/Accordion WithRef play-test 1`] = `
<div data-test-block="true">
  <div class
       id="dummyIdForwardedFromRef"
  >
    <div class="_accordionItem_ihaf2_1">
      <div class="_tag_ihaf2_11">
        <button class="_header_ihaf2_14  _header_medium_ihaf2_71"
                aria-expanded="false"
                type="button"
        >
          <div class="_titleWrapper_ihaf2_76">
            <div class="_title_ihaf2_76 _title_medium_ihaf2_100">
              Meg selv
            </div>
          </div>
          <div class="_iconWrapper_ihaf2_80 _iconWrapper_medium_ihaf2_86">
            <svg class="_systemIcon_dry7n_2 _systemIcon_large_dry7n_15 _icon_ihaf2_18 _chevron_medium_ihaf2_29 _chevron_closed_ihaf2_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M18.1 7.98001L12 14.08L5.9 7.98001L4 9.88L12 17.88L20 9.88L18.1 7.98001Z">
              </path>
            </svg>
          </div>
        </button>
      </div>
      <div class="_content_collapsed_ihaf2_120">
        Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/Accordion/Accordion WithSmall smoke-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="_accordionItem_ihaf2_1">
      <div class="_tag_ihaf2_11">
        <button class="_header_ihaf2_14"
                aria-expanded="false"
                type="button"
        >
          <div class="_titleWrapper_ihaf2_76">
            <div class="_title_ihaf2_76">
              Meg selv
            </div>
          </div>
          <div class="_iconWrapper_ihaf2_80">
            <svg class="_systemIcon_dry7n_2 _systemIcon_medium_dry7n_10 _icon_ihaf2_18  _chevron_closed_ihaf2_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M18.1 7.98001L12 14.08L5.9 7.98001L4 9.88L12 17.88L20 9.88L18.1 7.98001Z">
              </path>
            </svg>
          </div>
        </button>
      </div>
      <div class="_content_collapsed_ihaf2_120">
        Fikk du over 1 000 kroner i restskatt, deles summen opp i 2 fakturaer. Fristen for når du må betale avhenger av når du fikk skatteoppgjøret ditt.
      </div>
    </div>
  </div>
</div>
`;
